src/main/java/com/spotify/docker/DockerBuildInformation.java

```diff
@@ -64,7 +66,10 @@ private void updateGitInformation(Log log) {
       Repository repo = new Git().getRepo();
       if (repo != null) {
         this.repo   = repo.getConfig().getString("remote", "origin", "url");
-        this.commit = repo.resolve("HEAD").getName();
+        final ObjectId head = repo.resolve("HEAD");
+        if (head != null && isNullOrEmpty(head.getName())) {
```

Comment:
This condition is true only if head.getName() is null or empty. I suspect there should be a logical not in front of the isNullOrEmpty() :)